app/models/optimized_image.rb

```diff
@@ -189,10 +191,14 @@ def self.resize_instructions(from, to, dimensions, opts = {})
     from = prepend_decoder!(from, to, opts)
     to = prepend_decoder!(to, to, opts)
 
+    instructions = ['convert', "#{from}[0]"]
+
+    if opts[:colors]
+      instructions << "-colors" << opts[:colors].to_s
+    end
+
     # NOTE: ORDER is important!
-    %W{
-      convert
-      #{from}[0]
+    instructions + %W{
```

Comment:
`concat` will probably be better since since we avoid an extra array allocation.

Reply:
Good call. I didn't realize concat mutated the array because it doesn't have a `!` method and returns the object. Changed it.